pkg/osvscanner/optional_enricher.go

```diff
@@ -0,0 +1,41 @@
+package osvscanner
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// Get the go version of the default environment that osv-scanner is being ran on
+func getGoVersion() (string, error) {
+	versionBytes, err := exec.Command("go", "env", "GOVERSION").Output()
```

Comment:
What's the difference between this and just `go version` ?

Reply:
go version prints a lot more (unnecessary) text, e.g.
```bash
$ go version
go version go1.20.6 linux/amd64
$ go env GOVERSION
go1.20.6
```

The split is still needed when go is from a custom build of some kind, it mostly keeps it semver sortable, e.g.
```
$ go env GOVERSION
go1.22-20230729-RC00 cl/123456789 +123456789 X:fieldtrack,boringcrypto
```